keadm/cmd/keadm/app/cmd/beta/reset.go

```diff
@@ -0,0 +1,223 @@
+		Long:    resetLongDescription,
+		Example: resetExample,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			whoRunning, err := util.RunningModuleV2(reset)
+			if err != nil {
+				return err
+			}
+			switch whoRunning {
+			case common.KubeEdgeEdgeRunning:
+				IsEdgeNode = true
+			case common.NoneRunning:
+				fmt.Println("None of KubeEdge components are running in this host, exit")
+				os.Exit(0)
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !reset.Force {
+				fmt.Println("[reset] WARNING: Changes made to this host by 'keadm init' or 'keadm join' will be reverted.")
+				fmt.Print("[reset] Are you sure you want to proceed? [y/N]: ")
+				s := bufio.NewScanner(os.Stdin)
+				s.Scan()
+				if err := s.Err(); err != nil {
+					return err
+				}
+				if strings.ToLower(s.Text()) != "y" {
+					return fmt.Errorf("aborted reset operation")
+				}
+			}
+			// 1. kill cloudcore/edgecore process.
+			// For edgecore, don't delete node from K8S
+			if err := TearDownKubeEdge(IsEdgeNode, reset.Kubeconfig); err != nil {
+				return err
+			}
+
+			// 2. Remove containers managed by KubeEdge. Only for edge node.
+			if err := RemoveContainers(IsEdgeNode, utilsexec.New()); err != nil {
+				fmt.Printf("Failed to remove containers: %v\n", err)
+			}
+
+			// 3. Clean stateful directories
+			if err := cleanDirectories(IsEdgeNode); err != nil {
+				return err
+			}
+
+			// cleanup mqtt container
+			if err := RemoveMqttContainer(); err != nil {
+				fmt.Printf("Failed to remove MQTT container: %v\n", err)
+			}
+			//4. TODO: clean status information
+
+			return nil
+		},
+	}
+
+	addResetFlags(cmd, reset)
+	return cmd
+}
+
+func RemoveMqttContainer() error {
+	ctx := context.Background()
+
+	cli, err := dockerclient.NewClientWithOpts(dockerclient.FromEnv)
+	if err != nil {
+		return fmt.Errorf("init docker dockerclient failed: %v", err)
+	}
+
+	options := dockertypes.ContainerListOptions{}
+	options.Filters = dockerfilters.NewArgs()
+	options.Filters.Add("ancestor", "eclipse-mosquitto:1.6.15")
+
+	mqttContainers, err := cli.ContainerList(ctx, options)
+	if err != nil {
+		fmt.Printf("List MQTT containers failed: %v\n", err)
+		return err
+	}
+
+	for _, c := range mqttContainers {
+		err = cli.ContainerRemove(ctx, c.ID, dockertypes.ContainerRemoveOptions{RemoveVolumes: true, Force: true})
+		if err != nil {
+			fmt.Printf("failed to remove MQTT container: %v\n", err)
+		}
+	}
+
+	return nil
+}
+
+// TearDownKubeEdge will bring down either cloud or edge components,
+// depending upon in which type of node it is executed
+func TearDownKubeEdge(isEdgeNode bool, kubeConfig string) error {
+	var ke common.ToolsInstaller
+	ke = &util.KubeCloudHelmInstTool{
+		Common: util.Common{
+			KubeConfig: kubeConfig,
+		},
+	}
+	if isEdgeNode {
+		ke = &util.KubeEdgeInstTool{Common: util.Common{}}
+	}
+
+	err := ke.TearDown()
+	if err != nil {
```

Comment:
Does the ke.TearDown() remove the CRDs?

Reply:
no, just clean up `kubeedge` namespace, binary install mode also doesn't process CRDs